### PR TITLE
Migrate from app.php to IBootstrap

### LINKED ISSUE #353: Officeonline App "/appinfo/app.php is deprecated"
{"reqId":"wDCGQC42NaS7zEWbBtmg","level":0,"time":"2022-09-08T17:10:13+00:00","remoteAddr":"10.0.10.90","user":"admin","app":"officeonline","method":"GET","url":"/.well-known/webfinger","message":"/appinfo/app.php is deprecated, use \\OCP\\AppFramework\\Bootstrap\\IBootstrap on the application class instead.","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36","version":"25.0.0.9","data":{"app":"officeonline"},"id":"631a221985998"}``

### PR DESCRIPTION
**Pull request by patschi**
* Resolves: # 
* Target version: main

### Summary

Migrated from app.php to IBootstrap. Tested:
1. App is loaded
2. Settings can be changed
3. Documents can be opened

My PHP might be a bit rusty and PHP as well as Nextcloud seems to be evolving quite quickly - please check the commit thoroughly and let me know about any feedback. 

Fixes #353

### TODO

n/a

### Checklist

- [X] Code is properly formatted
- [X] Sign-off message is added to all commits
- [X] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: lib/Listener/LoadOfficeOnlineFilesActions.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+use OCA\Files\Event\LoadAdditionalScriptsEvent;
+use OCA\Officeonline\AppInfo\Application;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Util;
+
+class LoadOfficeOnlineFilesActions implements IEventListener {
+    public function handle(Event $event): void {
+        if (!($event instanceof LoadAdditionalScriptsEvent)) {
```
Comment (juliushaertl) [inline lib/Listener/LoadOfficeOnlineFilesActions.php:37]: Would be good to use the `LoadViewer` event here as before.

Comment (juliushaertl) [reply lib/Listener/LoadOfficeOnlineFilesActions.php:37]: Ah that is already present, I __think__ this one shouldn't be necessary, but let's keep it for now as this PR is just moving the code around.

**Review by juliushaertl — COMMENTED** [review]
Great, thanks a lot for your pull request and also the cleanup work on the issues. Looks good code wise but will give it a test run myself later this week.

File: lib/Listener/LoadOfficeOnlineFilesActions.php (hunk 1 above)
Comment (patschi) [reply lib/Listener/LoadOfficeOnlineFilesActions.php:37]: I pretty much oriented myself at https://github.com/nextcloud/contacts/pull/2364/files and https://github.com/nextcloud/contacts/blob/dc8bf5ac5ab10f25eff266f75f0396c494c6cdbb/lib/Listener/LoadContactsFilesActions.php#L30. So not sure what's the alternative to use

**Comment by joshtrichards** [comment]
Superseded by #490. Closing this out.

**Pull request closed without merging**